ethcore/private-tx/src/lib.rs

```diff
@@ -316,26 +383,26 @@ impl Provider where {
 						Ok(desc.received_signatures.len() + 1 == desc.validators.len())
 					}
 					false => {
-						trace!("Sender's state doesn't correspond to validator's");
+						warn!("Sender's state doesn't correspond to validator's");
 						bail!(ErrorKind::StateIncorrect);
 					}
 				}
 			}
 			Err(err) => {
-				trace!("Sender's state doesn't correspond to validator's, error {:?}", err);
+				warn!("Sender's state doesn't correspond to validator's, error {:?}", err);
 				bail!(err);
 			}
 		}
 	}
 
 	/// Broadcast the private transaction message to the chain
-	fn broadcast_private_transaction(&self, message: Bytes) {
-		self.notify(|notify| notify.broadcast(ChainMessageType::PrivateTransaction(message.clone())));
+	fn broadcast_private_transaction(&self, transaction_hash: H256, message: Bytes) {
+		self.notify(|notify| notify.broadcast(ChainMessageType::PrivateTransaction(transaction_hash, message.clone())));
```

Comment:
It looks suspect to `cloning` the vector here instead of moving into the closure! Any particular reason why?

Reply:
It's not possible to move captured outer variable (message)

Reply:
Just a note: in the future we may consider to change type `ChainMessageType` to only take reference of `[u8]`. In that case this should solve this message clone issue.